app/Jobs/SendEmptyWikiNotificationsJob.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Jobs;
+
+use App\Notifications\EmptyWikiNotification;
+use App\Wiki;
+use App\WikiNotificationSentRecord;
+use Carbon\CarbonImmutable;
+use Illuminate\Contracts\Queue\ShouldBeUnique;
+use Illuminate\Support\Facades\Log;
+
+class SendEmptyWikiNotificationsJob extends Job implements ShouldBeUnique
+{
+    public function handle (): void
+    {
+        $wikis = Wiki::with(['wikiLifecycleEvents'])->get();
+
+        foreach ($wikis as $wiki) {
+            try {
+                if ($this->checkIfWikiIsOldAndEmpty($wiki)) {
+                    $this->sendEmptyWikiNotification($wiki);
+                }
+            } catch (\Exception $exception) {
+                Log::error(
+                    'Failure processing wiki '.$wiki->getAttribute('domain').' for EmptyWikiNotification check: '.$exception->getMessage()
+                );
+                $this->fail();
+            }
+        }
+    }
+
+    public function checkIfWikiIsOldAndEmpty(Wiki $wiki)
+    {
+        // Calculate how many days has passed since the wiki instance was first created
+        $emptyDaysThreshold = config('wbstack.wiki_empty_notification_threshold');
+        $createdAt = $wiki->created_at;
+        $now = CarbonImmutable::now();
+        $emptyWikiDays = $createdAt->diffInDays($now);
+
+        $firstEdited = $wiki->wikiLifecycleEvents->first_edited;
+
+        $emptyWikiNotificationCount = WikiNotificationSentRecord::where([
+            'wiki_id' => $wiki->id,
+            'notification_type' => EmptyWikiNotification::TYPE
+        ])->count();
+
+        if (
+            $firstEdited == null &&
+            $emptyWikiDays >= $emptyDaysThreshold &&
+            $emptyWikiNotificationCount == 0
```

Comment:
Genuine question from someone who doesn't know PHP all to well: is it safe to use `==` instead of `===` here. Any chance this could go wrong?

Reply:
I think in this particular case it's safe since the `count` method appears to return a casted `int` and if something database related goes wrong I'd expect an Exception